File: library-core/src/main/java/com/mikepenz/fastadapter/IIdDistributor.java
package com.mikepenz.fastadapter;

import com.mikepenz.fastadapter.utils.DefaultIdDistributorImpl;

import java.util.List;

/**
 * Created by fabianterhorst on 16.07.17.
 */
public interface IIdDistributor<Identifiable extends IIdentifyable> {

    IIdDistributor<? extends IIdentifyable> DEFAULT = new DefaultIdDistributorImpl<>();

    List<Identifiable> checkIds(List<Identifiable> items);

    Identifiable[] checkIds(Identifiable... items);

    Identifiable checkId(Identifiable item);

    long nextId(Identifiable item);
}

File: library-core/src/main/java/com/mikepenz/fastadapter/IIdDistributor.kt
package com.mikepenz.fastadapter

import com.mikepenz.fastadapter.utils.DefaultIdDistributorImpl

/**
 * Created by fabianterhorst on 16.07.17.
 */
interface IIdDistributor<Identifiable : IIdentifyable> {

    fun checkIds(identifiables: List<Identifiable>): List<Identifiable>

    fun checkIds(vararg identifiables: Identifiable): Array<out Identifiable>

    fun checkId(identifiable: Identifiable): Identifiable

    fun nextId(identifiable: Identifiable): Long

    companion object {
        @JvmField
        val DEFAULT: IIdDistributor<out IIdentifyable> = DefaultIdDistributorImpl()
    }
}
